modules/core/04-channel/keeper/keeper_test.go

```diff
@@ -81,6 +82,73 @@ func (suite *KeeperTestSuite) TestGetAppVersion() {
 	suite.Require().Equal(ibcmock.Version, channelVersion)
 }
 
+// TestGetAllChannelsWithPortPrefix verifies ports are filtered correctly using a port prefix.
+func (suite *KeeperTestSuite) TestGetAllChannelsWithPortPrefix() {
+	tests := []struct {
+		name             string
+		prefix           string
+		allChannels      []types.IdentifiedChannel
+		expectedChannels []types.IdentifiedChannel
+	}{
+		{
+			name:   "transfer channel is retrieved with prefix",
+			prefix: "tra",
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel("different-portid", "channel-1", types.Channel{}),
+			},
+			expectedChannels: []types.IdentifiedChannel{types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{})},
+		},
+		{
+			name:   "matches port with full name as prefix",
+			prefix: transfertypes.TypeMsgTransfer,
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, "channel-0", types.Channel{}),
+				types.NewIdentifiedChannel("different-portid", "channel-1", types.Channel{}),
+			},
+			expectedChannels: []types.IdentifiedChannel{types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{})},
+		},
+		{
+			name:   "no ports match prefix",
+			prefix: "wont-match-anything",
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel("different-portid", "channel-1", types.Channel{}),
+			},
+			expectedChannels: nil,
+		},
+		{
+			name:   "empty prefix matches everything",
+			prefix: "",
+			allChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel("different-portid", "channel-1", types.Channel{}),
+			},
+			expectedChannels: []types.IdentifiedChannel{
+				types.NewIdentifiedChannel(transfertypes.TypeMsgTransfer, ibctesting.FirstChannelID, types.Channel{}),
+				types.NewIdentifiedChannel("different-portid", "channel-1", types.Channel{}),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			for _, ch := range tc.allChannels {
+				suite.chainA.GetSimApp().GetIBCKeeper().ChannelKeeper.SetChannel(suite.chainA.GetContext(), ch.PortId, ch.ChannelId, types.Channel{})
+			}
```

Comment:
Maybe we should just preset a transfer channel and a different portID channel for all test cases? Then just change the prefix and the expected channels?

Reply:
personally I think it's more readable to have the actual and expected values side by side in a given test case, without needing to look down at the test to see which channels are added there as initial values.

There is some duplication this way, but I would lean in favour of readability vs de-deduplication for a test suite like this.

WDYT?

Reply:
That makes sense to me :+1: The only other change I might do is:
```go
allChannels: []types.IdentifiedChannel{
	types.NewIdentifiedChannel(transferChannel),
	types.NewIdentifiedChannel(differentPortIDChannel),
},
expectedChannels: []types.IdentifiedChannel{
	types.NewIdentifiedChannel(transferChannel}),
	types.NewIdentifiedChannel(differentPortIDChannel),
},
```
			
Just so that there is less to read